### PR TITLE
Update useReferenceInputController.ts

### PR DESCRIPTION
**Pull request by Ferlorin**
## Problem

Error while building:

`node_modules/ra-core/src/controller/input/useReferenceInputController.ts:121:13 - error TS2322: Type 'Enabled<{ data: any[]; total?: number; pageInfo?: { hasNextPage?: boolean; hasPreviousPage?: boolean; }; }, Error, { data: any[]; total?: number; pageInfo?: { hasNextPage?: boolean; hasPreviousPage?: boolean; }; }, QueryKey>' is not assignable to type 'Enabled<RecordType[], Error, RecordType[], QueryKey>'.
  Type '(query: Query<{ data: any[]; total?: number; pageInfo?: { hasNextPage?: boolean; hasPreviousPage?: boolean; }; }, Error, { data: any[]; total?: number; pageInfo?: { hasNextPage?: boolean; hasPreviousPage?: boolean; }; }, QueryKey>) => boolean' is not assignable to type 'Enabled<RecordType[], Error, RecordType[], QueryKey>'.
    Type '(query: Query<{ data: any[]; total?: number; pageInfo?: { hasNextPage?: boolean; hasPreviousPage?: boolean; }; }, Error, { data: any[]; total?: number; pageInfo?: { hasNextPage?: boolean; hasPreviousPage?: boolean; }; }, QueryKey>) => boolean' is not assignable to type '(query: Query<RecordType[], Error, RecordType[], QueryKey>) => boolean'.
      Types of parameters 'query' and 'query' are incompatible.
        Type 'Query<RecordType[], Error, RecordType[], QueryKey>' is not assignable to type 'Query<{ data: any[]; total?: number; pageInfo?: { hasNextPage?: boolean; hasPreviousPage?: boolean; }; }, Error, { data: any[]; total?: number; pageInfo?: { hasNextPage?: boolean; hasPreviousPage?: boolean; }; }, QueryKey>'.
          Property 'data' is missing in type 'RecordType[]' but required in type '{ data: any[]; total?: number; pageInfo?: { hasNextPage?: boolean; hasPreviousPage?: boolean; }; }'.

121             enabled: currentValue != null && currentValue !== '',
                ~~~~~~~
`

## Solution

Define the type of the newly crated object

## How To Test

build the project

### CODE REVIEW

File: packages/ra-core/src/controller/input/useReferenceInputController.ts [hunk 1]
```diff
@@ -121,7 +121,7 @@ export const useReferenceInputController = <RecordType extends RaRecord = any>(
             enabled: currentValue != null && currentValue !== '',
             meta,
             ...otherQueryOptions,
-        },
+        } as UseQueryOptions<RecordType[], Error>,
```
Comment (slax57) [inline packages/ra-core/src/controller/input/useReferenceInputController.ts:124]: If this is supposed to fix the TS type, can you please remove the `@ts-ignore` at line 119 which should no longer be necessary?
Thanks.

**Review by slax57 — CHANGES_REQUESTED** [review]
Thank you for the PR.

However I have doubts this will fix anything, since, as I mentioned, there is still a `@ts-ignore` on the same object. Besides, we build the project several times a day, both on the CI and on our local machines, and never had the error you mention.
You may need to investigate the error a little further.

File: packages/ra-core/src/controller/input/useReferenceInputController.ts (hunk 1 above)
Comment (Ferlorin) [reply packages/ra-core/src/controller/input/useReferenceInputController.ts:124]: When I say fail to build. i was getting errors when building my project with npm run build. However I re-installed everything from the beginning and a new .eslintrc file was created and it worked this time. However these issues are going red in the IDE when using typescript

**Comment by slax57** [comment]
You need to use `yarn` to build the react-admin packages, otherwise you are completely bypassing the `yarn.lock` file, which is probably the root issue why your IDE is complaining.
Feel free to reopen if that does not fix the issue.

**Pull request closed without merging**